frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts

```diff
@@ -385,28 +321,170 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
+        
+        // if the selected option is a simple metric and it was earlier a repeated metrics, we will clear the keys and set selectedNode
+        if (keyIndex == 0 && this.getKeys(queryIndex).length > 1) {
+          this.getKeys(queryIndex).clear();
+          this.addMoreSelectKey(event.option.value.key, queryIndex, keyIndex);
+          let metric = this.allMetrics.find(metric => metric.key === event.option.value.key);
+          this.firstSelectedNode[queryIndex] = metric;
+          this.selectedNode[queryIndex] = metric;
+          const { metadata: { type } } = this.selectedNode[queryIndex];
+          this.filteredStatistic$[queryIndex] = this.setFilteredStatistic(type);
+          // set editMode to true to avoid isMetricRepeated() function to take experimentInfo instead of current form changes
+          this.editMode = true;
+        } else {
+          // if the selected option is a simple metric and it was earlier not a repeated metrics, set selectedNode
+          const keys = this.getKeys(queryIndex).getRawValue();
+          this.selectedNode[queryIndex] = keys[keys.length - 1].metricKey;
+          const { metadata: { type } } = this.selectedNode[queryIndex];
+          this.filteredStatistic$[queryIndex] = this.setFilteredStatistic(type);
+        }
+        // handling leaf node of repeated metrics
+        let metric = this.allMetrics.find(metric => metric.key === event.option.value.key);
+        this.selectedNode[queryIndex] = metric;
+        // reset options for metric keys:
+        this.optionsSub();
+      }
+    }
+  }
+
+  updateView(type?: string) {
+    this.metricsDataSource.next(this.queries.controls);
+    if (type) {
+      this[type].nativeElement.scroll({
+        top: this[type].nativeElement.scrollHeight,
+        behavior: 'smooth'
+      });
+    }
+  }
+
   emitEvent(eventType: NewExperimentDialogEvents) {
     switch (eventType) {
       case NewExperimentDialogEvents.CLOSE_DIALOG:
         this.emitExperimentDialogEvent.emit({ type: eventType });
         break;
       case NewExperimentDialogEvents.SEND_FORM_DATA:
       case NewExperimentDialogEvents.SAVE_DATA:
-        this.queryNameError = [];        
+        this.queryMetricKeyError = [];
+        this.queryStatisticError = []; 
+        this.queryComparisonStatisticError = [];
+        this.queryNameError = [];
         const monitoredMetricsFormData = this.queryForm.getRawValue();
         monitoredMetricsFormData.queries = monitoredMetricsFormData.queries.map(
           (query, index) => {
             let { keys, operationType, queryName, compareFn, compareValue, repeatedMeasure } = query;
             if (keys) {
-              keys = keys.filter((key) => key.metricKey !== null).map(key => key.metricKey.key);
+              keys = keys.filter((key) => key.metricKey !== null).map(key => key.metricKey.key ? key.metricKey.key : key.metricKey);
```

Comment:
what if `key.metricKey` is `undefined` or other falsey values? It will only filter out if it's exactly `null` here.

Reply:
We push `null` for metrickey on adding a new query, so I would have filtered with `null`